compiler/passes/docs.cpp

```diff
@@ -295,6 +295,76 @@ std::string generateSphinxProject(std::string dirpath) {
   return std::string(moddir);
 }
 
+/* Checks for the valid Project Version (non-recursively). If an error occurs, exit and report
+ * error.
+ */
+static char * checkProjectVersion(char * projectVersion) {
+  int length = strlen(projectVersion);
+  int i = 0;
+  int dot = 0;
+  bool check = true;
+  int tagIndex = 0;
+
+  // Supported version tags
+  const char * tags[] = {"alpha", "beta", "rc"};
+  const char * error = "";
+  for(i=0; i<length; i++) {
+    if(i>0 && projectVersion[i] == '.') {
+      if(projectVersion[i-1] != '.') {
+        dot++;
+        if(dot>2) {
+          error = "Required only two dots which seperates three numbers";
+          check = false;
+          break;
+        }
+        if(i == length-1) {
+          error = "Cannot end with dot, can end with either number or tag";
+          check = false;
+          break;
+        }
+      } else {
+        error = "Missing number between dots";
+        check = false;
+        break;
+      }
+    } else if(projectVersion[i] == '-' && dot == 2) {
+      if(projectVersion[i-1] != '.') {
+        tagIndex = i+1;
+        break;
+      } else {
+        error = "Missing number before tag";
+        check = false;
+        break;
+      }
+    } else if(projectVersion[i] == '-' && dot != 2) {
+      error = "Required only two dots which seperates three numbers";
+      check = false;
+      break;
+    } else if((int)projectVersion[i] > (int)'9' || (int)projectVersion[i] < (int)'0') {
+      error = "Invalid Characters, only digits, dot and hyphen permitted";
+      check = false;
+      break;
+    }
+  }
+  if(dot != 2 && i == length) {
+    error = "Required two dots which seperates three numbers";
+    check = false;
+  }
+  if(check && tagIndex>0) {
+    if(!(strcmp(projectVersion+tagIndex,tags[0]) == 0 || strcmp(projectVersion+tagIndex,tags[1]) == 0 || strcmp(projectVersion+tagIndex,tags[2]) == 0)) {
```

Comment:
Minor nit, we try to keep lines less than 80 characters for legibility.  I would break this up something like this:
```C
if(!(strcmp(projectVersion+tagIndex,tags[0]) == 0 ||
   strcmp(projectVersion+tagIndex,tags[1]) == 0 ||
   strcmp(projectVersion+tagIndex,tags[2]) == 0)) {
```

Reply:
It would also be reasonable to iterate through the tags array, instead of hard-coding the number of supported tags.